### PR TITLE
95 - Need error message if user does not have correct roles

### PR DESCRIPTION
**Pull request by nlaprell**
- Updated login messaging when no queries have been created:
    - Messaging appears in a yellow "warning" alert box.
    - The wizard user is directed to the crud page (with a hyperlink) to create queries.
    - The search user is still directed to have a wizard user create queries.
- The search button on the navigation is now hidden if there are no queries to search.
    - The search button appears as soon as a query is created.
- The wizard user now has a data-explorer-search-role as well (fixed search errors for the wizard user)
- Crud get requests now require the data-explorer-search-role instead of the data-explorer-wizard-role since they are used for searching.

### CODE REVIEW

File: src/main/ml-modules/root/client/app/account/login/login.controller.js [hunk 1]
```diff
@@ -17,9 +17,12 @@ angular.module('demoApp')
         .then( function() {
           $http.get('/api/checkTemplates').success(function(data, status, headers, config) {
             if (status == 200) {
-              if(!data.queryTemplateExists && data.isSearchUser) {
-                Auth.homeMessage = "There are no queries to search. Please contact the Data Explorer admin (Wizard User) to create queries";
-                Auth.homeMessageClass = "alert alert-warning";
+              if(!data.queryTemplateExists) {
+                if(data.isSearchUser) {
+                  Auth.homeMessage = "<span class=\"alert alert-warning\">There are no queries to search. Please contact the Data Explorer admin (Wizard User) to create queries.</span>";
```
Comment (damonfeldman) [inline src/main/ml-modules/root/client/app/account/login/login.controller.js]: People may not know about the ML security role of wizard user - suggest rewording to

 "Please contact a query maintainer who can add new queries that will show here. These users are anyone configured with the 'wizard-user' role in MarkLogic."

Comment (nlaprell) [reply src/main/ml-modules/root/client/app/account/login/login.controller.js]: Good call. I've pushed the suggested rewording.

**Comment by cskeefer** [comment]
All the updated messages and hiding the search menu look good. I feel like the changes to the following files to check for is-search-user rather than is-wizard-user would allow a search user to call the end points (by pasting in the URL in the browser). Since those are used Edit config menu, should we be checking for is-wizard-user instead?
- src/main/ml-modules/root/server/endpoints/api-crud-get-query-view.xqy
- src/main/ml-modules/root/server/endpoints/api-crud-list-queries.xqy
- src/main/ml-modules/root/server/endpoints/api-crud-list-views.xqy

**Comment by nlaprell** [comment]
@cskeefer - Those three endpoints are actually GET endpoints and do not perform any CRUD operations. The search page uses them when selecting a database, query, and view to search, so if we validate against the wizard role, it breaks for the search user when searching.

In testing, I manually pasted in the crud page into the browser, and while it does display the interface, any CRUD operation on that page immediately throws a 401 as expected and no data is changed. Eventually, we should probably throw an error on the page and disable the forms, but it works as is right now.
